workitem/workitem_repository_blackbox_test.go

```diff
@@ -322,3 +322,47 @@ func (s *workItemRepoBlackBoxTest) TestLookupIDByNamedSpaceAndNumberNotFound() {
 	require.NotNil(s.T(), err)
 	assert.IsType(s.T(), errors.NotFoundError{}, errs.Cause(err))
 }
+
+func (s *workItemRepoBlackBoxTest) TestLookupIDByNamedSpaceAndNumberStaleSpace() {
+	testFxt := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1), tf.WorkItems(20, func(testf *tf.TestFixture, idx int) error {
+		testf.WorkItems[idx].Type = workitem.SystemBug
+		testf.WorkItems[idx].Fields[workitem.SystemTitle] = fmt.Sprintf("title %d", idx)
+		testf.WorkItems[idx].Fields[workitem.SystemState] = workitem.SystemStateNew
+		testf.WorkItems[idx].Fields[workitem.SystemCreator] = testf.Identities[0].ID.String()
+		return nil
+	}))
+	sp := *testFxt.Spaces[0]
+	wi := *testFxt.WorkItems[0]
+	in := *testFxt.Identities[0]
+	wiID, spaceID, err := s.repo.LookupIDByNamedSpaceAndNumber(s.Ctx, in.Username, sp.Name, wi.Number)
+	require.Nil(s.T(), err)
+	require.NotNil(s.T(), wiID)
+	assert.Equal(s.T(), wi.ID, *wiID)
+	require.NotNil(s.T(), spaceID)
+	assert.Equal(s.T(), wi.SpaceID, *spaceID)
+
+	// delete above space
+	spaceRepo := space.NewRepository(s.DB)
+	err = spaceRepo.Delete(s.Ctx, sp.ID)
+	require.Nil(s.T(), err)
+
+	testFxt2 := tf.NewTestFixture(s.T(), s.DB, tf.Spaces(1, func(testf *tf.TestFixture, idx int) error {
+		testf.Spaces[0].Name = sp.Name
+		testf.Spaces[0].OwnerId = in.ID
+		return nil
+	}), tf.WorkItems(20, func(testf *tf.TestFixture, idx int) error {
+		testf.WorkItems[idx].Type = workitem.SystemBug
```

Comment:
Again, please don't specify a type explicitly.

Reply:
Fixed in https://github.com/fabric8-services/fabric8-wit/pull/1643/commits/5f826703ded88dc2c908170876d5982e276daece